projects/packages/publicize/src/rest-api/class-connections-controller.php

```diff
@@ -165,23 +177,35 @@ public function get_item_schema() {
 	/**
 	 * Get all connections. Meant to be called directly only on WPCOM.
 	 *
-	 * @param bool $run_tests Whether to run tests on the connections.
+	 * @param array $args Arguments
+	 *                    - 'test_connections': bool Whether to run connection tests.
+	 *                    - 'include': enum('site-wide', 'shared') Which connections to include.
 	 *
 	 * @return array
 	 */
-	protected static function get_all_connections( $run_tests = false ) {
+	protected static function get_all_connections( $args = array() ) {
 		/**
 		 * Publicize instance.
-		 *
-		 * @var \Automattic\Jetpack\Publicize\Publicize $publicize
 		 */
 		global $publicize;
 
 		$items = array();
 
+		$run_tests = $args['test_connections'] ?? false;
+		$include   = $args['include'] ?? '';
+
 		$test_results = $run_tests ? self::get_connections_test_status() : array();
 
-		foreach ( (array) $publicize->get_services( 'connected' ) as $service_name => $connections ) {
+		if ( 'site-wide' === $include ) {
```

Comment:
Can we do this by determining the context? If it's a user request, we return the connections for the user, and if it's blog request, we return the site's connections.

Reply:
Yeah, I think we can do that. 🤔

Reply:
I looked into it and there is one small issue with that.

When the request is made to the Jetpack site's REST API, how do we decide what the user intends to receive?

For example, in future, if we want the site admins to manage all the connections on their site, and not just the shared connections, then in that case we will simply call

```ts
await wp.apiFetch({
    path: '/wpcom/v2/publicize/connections'
})
```
which will always return the connections for the current user, along with the shared ones.

So, we need a way to have to decide which connections to return in that case.

Reply:
I have removed the scope param from the API args